handbook/network-security/network-security.md

```diff
@@ -131,6 +131,42 @@ When you have received a certificate from an certificate signing authority, you
 !!! note "TLS certificates with application load balancers"
     If you have a load balancer in front of your SystemLink application server you must ensure the same certificate is installed on both the load balancer and NI Web Server for targets to successfully connect and publish data to SystemLink.
 
+#### Specifying custom Diffie-Helman parameters
+
+The TLS protocol typically uses the [Diffie-Hellman key exchange algorithm](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange) as part of its connection handshake. The Diffie-Hellman algorithm requires the server to provide a prime number as an input to the algorithm. By default, NI Web Server uses a set of default prime values defined by the Apache Web Server for this purpose. This approach is sufficient for many use cases, but, when additional security is required, a custom value can be provided in place of the defaults.
+
+These instructions use the copy of OpenSSL installed with SystemLink 23.5 or later. A different copy of OpenSSL can be substituted if desired.
+
+1. Open a command prompt on the SystemLink Server machine.
+
+1. Run
+
+        "c:\Program Files\National Instruments\Shared\Skyline\OpenSSL\openssl.exe dhparam -outform PEM -out dhparam.txt <numbits>
+
+    The `<numbits>` value specifies a bit length for the prime. The bit length must be one of 1024, 2048, 3072, 4096, 7680, or 8192 bits. A length of at least 3072 bits is [strongly recommended](https://www.keylength.com/en/4/).
+
+1. OpenSSL will run for some time and produce a file call `dhparam.txt` in the current directory. Open this file in Notepad or another preferred text editor.
+
+1. Run Notepad as an administrator. Open the NI Web Server certificate. This file is located at `C:\Program Files\National Instruments\Shared\Web Server\certs\`. The actual name of the certificate file will vary per deployment.
```

Comment:
> The actual name of the certificate file will vary per deployment.

It's also possible to have multiple certificates in that directory (e.g., the user replaced the certificate and chose the "Leave the certificate in its current location" option). There are a couple ways to figure out which certificate is the right one:

1. Look in `C:\Program Files\National Instruments\Shared\Web Server\conf\defines.d\50_httpd-defines.conf` at the `TLS_CERTIFICATE_PATH` variable
2. On the HTTPS tab of the NI Web Server config utility, select to disable HTTPS, click the "Uninstall the certificate" link that appears at the top, click the "Apply and restart" button, select the "Leave the certificate in its current location" option, then click the link showing the certificate location. At this point, you'll want to cancel the dialog and then click the revert button (or simply exit the application).

Not sure if we care to document either approach.

Reply:
I added a note with the first option.